File: questao_06/questao_06.java
class Motor {
    private String tipo;

    public Motor(String tipo) {
        this.tipo = tipo;
    }

    public String getTipo() {
        return tipo;
    }
}

class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private Motor motor;

    public Carro(String marca, String modelo, int ano, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.motor = motor;
    }

    
    public void exibirDetalhes() {
        System.out.printf("Carro: %s %s, Ano: %d, Motor: %s%n", marca, modelo, ano, motor.getTipo());
    }

    public static void main(String[] args) {
        Motor motor1 = new Motor("Gasolina");
        
        Carro carro1 = new Carro("Peugeot", "208", 2021, motor1);
        
        carro1.exibirDetalhes();
    }
}

File: questao_06/questao_06.go
package main

import (
	"fmt"
)

type Motor struct {
	tipo string
}

func (m Motor) GetTipo() string {
	return m.tipo
}

type Carro struct {
	marca string
	modelo string
	ano int
	motor Motor
}

func (c Carro) ExibirDetalhes() {
	fmt.Printf("Carro: %s %s, Ano: %d, Motor: %s\n", c.marca, c.modelo, c.ano, c.motor.GetTipo())
}

func main() {
	motor1 := Motor{tipo: "Gasolina"}
	
	carro1 := Carro{marca: "Peugeot", modelo: "208", ano: 2021, motor: motor1}
	
	carro1.ExibirDetalhes()
}
